pkg/didcomm/protocol/introduce/models.go

```diff
@@ -0,0 +1,106 @@
+	Where           string          `json:"where,omitempty"`
+	ImgAttach       ImgAttach       `json:"img~attach,omitempty"`
+	Proposed        bool            `json:"proposed,omitempty"`
+}
+
+// DescriptionL10N may contain locale field and key->val pair for translation
+// e.g { "locale": "en", "es": "Donde se toma el MRI; no en el centro"},
+// where locale field tells that field Description form To struct has en translation
+type DescriptionL10N map[string]string
+
+// Locale returns locale for the specified description (To.Description)
+func (d DescriptionL10N) Locale() string {
+	if d == nil {
+		return ""
+	}
+	// TODO: clarify whether it should be default locale e.g "en" or empty string
+	return d["locale"]
+}
+
+// ImgAttach represent information about the image
+type ImgAttach struct {
+	Description string  `json:"description,omitempty"`
+	MimeType    string  `json:"mime-type,omitempty"`
+	Filename    string  `json:"filename,omitempty"`
+	Content     Content `json:"content,omitempty"`
+}
+
+// Content keeps image data
+type Content struct {
+	Link      string `json:"link,omitempty"`
+	ByteCount int    `json:"byte_count,omitempty"`
+	Sha256    string `json:"sha256,omitempty"`
+}
+
+// Timing keeps expiration time for Proposal
+type Timing struct {
+	ExpiresTime time.Time `json:"expires_time,omitempty"`
+}
+
+// PleaseIntroduceTo includes all field from To structure
+// also it has Discovered the field which should be provided by help-me-discover protocol
+type PleaseIntroduceTo struct {
+	To
+	//Discovered    Discovered `json:"discovered,omitempty"`
+}
+
+// Request is not part of any state machine, it can be sent at any time,
+// and when it is received, the recipient can choose whether or not to honor it in their own way
+// TODO: need to clarify about decorator ~please_ack and problem_report
+// 		 should Request contain those fields? What type it should be for each field?
+type Request struct {
+	Type              string            `json:"@type,omitempty"`
+	ID                string            `json:"@id,omitempty"`
+	PleaseIntroduceTo PleaseIntroduceTo `json:"please_introduce_to,omitempty"`
+	NWise             bool              `json:"nwise,omitempty"`
+	Timing            Timing            `json:"~timing,omitempty"`
+}
+
+// Response message that introducee usually sends in response to an introduction proposal
+type Response struct {
+	Type       string                  `json:"@type,omitempty"`
+	ID         string                  `json:"@id,omitempty"`
+	Thread     *decorator.Thread       `json:"~thread,omitempty"`
+	Approve    bool                    `json:"approve,omitempty"`
+	Invitation *didexchange.Invitation `json:"invitation,omitempty"`
+}
+
+// Ack confirming to an introducee which let them know that their invitation was forwarded
+// TODO: Ack was copy-pasted from didexchange clarify what status does.
+//       Do we need status for introduce protocol? What value it can be e.g "success", "error" ?
+type Ack struct {
```

Comment:
We should move this to something like a "common" package under the existing `didcomm` one. This same structure is used by the did-exchange protocol as well (and is reused in several other RFCs).

Reply:
I moved common model to  `pkg/didcomm/common/model`